t/bib_rights.t

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+BEGIN {
+  unshift(@INC, $ENV{'SDRROOT'}. '/crms/cgi');
+  unshift(@INC, $ENV{'SDRROOT'}. '/crms/post_zephir_processing');
+}
+
+use utf8;
+binmode(STDOUT, ':encoding(UTF-8)');
+use CRMS;
+use Encode;
+use JSON::XS;
+use bib_rights;
+use MARC::File::XML(BinaryEncoding => 'utf8');
+use Test::More;
+
+my $crms = CRMS->new();
+my $fixtures_dir = $ENV{'SDRROOT'} . '/crms/t/fixtures/bib_rights/';
+my $test_struct = read_json('bib_rights_tests.json');
+  
+foreach my $htid (sort keys %$test_struct) {
+  run_tests_for_htid($htid);
+}
+
+### MISCELLANEOUS COVERAGE TESTS
+delete $ENV{BIB_RIGHTS_DATE};
+
+# Pass nonexistent fed docs file
+$ENV{'us_fed_pub_exception_file'} = $fixtures_dir . 'no_such_file.txt';
+my $br = bib_rights->new();
+# Clone existing BR object
+ok(my $br2 = $br->new());
+
+delete $ENV{'us_fed_pub_exception_file'};
+
+my $date = bib_rights::getDate(0);
+is($date, '19700101:00:00:00');
+
+my $fake_marc_xml = '<record><leader>03895cas a2200517I  4500</leader></record>';
+my $fake_marc = MARC::Record->new_from_xml($fake_marc_xml);
+my $bib_info = $br->get_bib_info($fake_marc, 'BOGUS RECORD 2');
+ok(!scalar keys %$bib_info);
```

Comment:
Some options for structuring:

- As you know I'm fond of `Test::Spec`, which lets us structure these in an rspec-like style
- But if you don't want to do that, "subtest" from `Test::More`: https://perldoc.perl.org/Test::More#subtest gives a way to structure some without adding additional dependencies

Reply:
I can go for `subtest`. I like Test::More but it is awfully hard to structure.